core/src/nodes/handled_node_tasks.rs

```diff
@@ -316,7 +320,24 @@ impl<TInEvent, TOutEvent, TIntoHandler, TReachErr, THandlerErr, TUserData, TPeer
     // TODO: look into merging with `poll()`
     fn poll_inner(&mut self) -> Async<(InToExtMessage<TOutEvent, TIntoHandler, TReachErr, THandlerErr, TPeerId>, TaskId)> {
         for to_spawn in self.to_spawn.drain() {
-            tokio_executor::spawn(to_spawn);
+            // We try to use the default executor, but fall back to polling the task manually if
+            // no executor is available. This makes it possible to use the core in environments
+            // outside of tokio.
+            let mut executor = tokio_executor::DefaultExecutor::current();
```

Comment:
How about using the `futures::future::Executor` trait instead as an additional parameter type when constructing `HandledNodesTasks`? You could provide an implementation that loops over a set of futures or else users could plug in their own alternatives, including using a tokio executor.

Reply:
Although it would be difficult to only use the small `Executor` interface. It can not accommodate the manual looping you are doing here I guess.

Reply:
We discussed this together, and concluded that it's a change orthogonal to this PR.